### PR TITLE
GCP SparkR operator

### PR DESCRIPTION
**Pull request by roelhogervorst**
This PR adds a spark_R operator which will allow you to schedule R, and sparkR jobs on a dataproc cluster. The functionality to run that kind of job is already in dataproc, but for some reason there is no operator in Airflow.

---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/contrib/operators/dataproc_operator.py [hunk 1]
```diff
@@ -157,6 +157,22 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
 
+class DataProcSparkROperator(DataprocSubmitSparkRJobOperator):
```
Comment (mik-laj) [inline airflow/contrib/operators/dataproc_operator.py]: It is not necessary. This is only needed if this operator was available in Airflow 1.10.x

Comment (roelhogervorst) [reply airflow/contrib/operators/dataproc_operator.py]: merci, will remove

File: airflow/providers/google/cloud/operators/dataproc.py [hunk 2]
```diff
@@ -1359,6 +1359,128 @@ def execute(self, context):
         super().execute(context)
 
 
+class DataprocSubmitSparkRJobOperator(DataprocJobBaseOperator):
```
Comment (turbaszek) [inline airflow/providers/google/cloud/operators/dataproc.py:1362]: Do we need custom operator for R jobs? Can't we use the `DataprocSubmitJobOperator`?

Comment (turbaszek) [reply airflow/providers/google/cloud/operators/dataproc.py:1362]: The `DataprocJobBaseOperator` as well as other XJobOperators will be deleted in future:
```python
        # TODO: Remove one day
        warnings.warn(
            "The `{cls}` operator is deprecated, please use `DataprocSubmitJobOperator` instead. You can use"
            " `generate_job` method of `{cls}` to generate dictionary representing your job"
            " and use it with the new operator.".format(cls=type(self).__name__),
            DeprecationWarning,
            stacklevel=1
        )
```

**Comment by turbaszek** [comment]
Hi @roelhogervorst any progress? I am happy to help with running system tests for the new SparkR job :)

**Comment by roelhogervorst** [comment]
I'm so sorry @turbaszek , I completely missed your message. Are you saying we can submit jobs without specifying the type of job?

File: airflow/providers/google/cloud/operators/dataproc.py (hunk 2 above)
Comment (roelhogervorst) [reply airflow/providers/google/cloud/operators/dataproc.py:1362]: Is there an example of how to use the generic DataprocJobBaseOperator for all the jobs yet, @turbaszek?

Comment (mik-laj) [reply airflow/providers/google/cloud/operators/dataproc.py:1362]: https://github.com/apache/airflow/blob/master/airflow/providers/google/cloud/example_dags/example_dataproc.py#L141-L166

Comment (roelhogervorst) [reply airflow/providers/google/cloud/operators/dataproc.py:1362]: oh that is way easier, I'll create a new PR

**Pull request closed without merging**